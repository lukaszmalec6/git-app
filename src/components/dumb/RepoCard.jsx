import React, { Component } from "react";
import Button from "../dumb/Button.jsx";
import Colors from "../dumb/LangColors.json";
import PropTypes from "prop-types";
export default class RepoCard extends Component {
  render() {
    const { name, description, lang, url, stars, watchers } = this.props;
    return (
      <section className="repo-card-container">
        <div>
          <div className="desc">
            <a href={url}>
              <h1>{name}</h1>
              <h2>{description}</h2>
            </a>
            <div className="details">
              <span style={{ backgroundColor: Colors[lang] }} />
              <h3>{lang ? lang : "Unspecified lang"}</h3>
              <span />
              <h3>Stars: {stars}</h3>
              <span />
              <h3>Watchers: {watchers}</h3>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
RepoCard.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  lang: PropTypes.string,
  url: PropTypes.string.isRequired,
  stars: PropTypes.number.isRequired,
  watchers: PropTypes.number.isRequired
};
