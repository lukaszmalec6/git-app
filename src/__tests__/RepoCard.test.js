import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import RepoCard from "../components/dumb/RepoCard.jsx";
describe("RepoCard", () => {
  it("should render correctly", () => {
    const output = shallow(
      <RepoCard
        name="something"
        description="something"
        lang="something"
        url="something"
        stars={512}
        watchers={1024}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
