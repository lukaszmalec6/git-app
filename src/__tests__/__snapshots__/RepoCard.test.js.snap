// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard should render correctly 1`] = `
<section
  className="repo-card-container"
>
  <div>
    <div
      className="desc"
    >
      <a
        href="something"
      >
        <h1>
          something
        </h1>
        <h2>
          something
        </h2>
      </a>
      <div
        className="details"
      >
        <span
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <h3>
          something
        </h3>
        <span />
        <h3>
          Stars: 
          512
        </h3>
        <span />
        <h3>
          Watchers: 
          1024
        </h3>
      </div>
    </div>
  </div>
</section>
`;
